app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -58,6 +65,24 @@ def finalize(*args)
         check_exit_status
       end
 
+      def self.feature_job_event_name(label, suffix = :success)
+        ::Foreman::Observable.event_name_for("#{::Actions::RemoteExecution::RunHostJob.event_name_base}_#{label}_#{::Actions::RemoteExecution::RunHostJob.event_name_suffix(suffix)}")
+      end
+
+      def emit_feature_event(execution_plan, hook = :success)
+        return unless root_action?
+
+        if input["job_features"]
+          input['job_features'].each do |feature|
+            name = "#{self.class.event_name_base}_#{feature}_#{self.class.event_name_suffix(hook)}"
+            trigger_hook name, payload: event_payload(execution_plan)
```

Comment:
In case there are multiple features, could we build the event payload just once?

Reply:
Absolutely.